lib/auth/windows/ldap.go

```diff
@@ -167,6 +181,47 @@ func (c *LDAPClient) Close() {
 	c.mu.Unlock()
 }
 
+func extractReferrals(err error) []string {
+	if err == nil {
+		return nil
+	}
+
+	var ldapErr *ldap.Error
+	if !errors.As(err, &ldapErr) {
+		return nil
+	}
+	if ldapErr.ResultCode != ldap.LDAPResultReferral {
+		return nil
+	}
+	searchResultIndex := slices.IndexFunc(ldapErr.Packet.Children, func(packet *ber.Packet) bool {
+		return packet.Description == "Search Result Done"
+	})
+	if searchResultIndex < 0 {
+		return nil
+	}
+	searchResult := ldapErr.Packet.Children[searchResultIndex]
+
+	referralsIndex := slices.IndexFunc(searchResult.Children, func(packet *ber.Packet) bool {
+		return packet.Description == "Referral"
+	})
+	if referralsIndex < 0 {
+		return nil
+	}
+	referrals := searchResult.Children[referralsIndex].Children
+
+	out := make([]string, 0, len(referrals))
+	for _, referral := range referrals {
+		referralValue := referral.Value.(string)
+		last := strings.LastIndex(referralValue, "/")
```

Comment:
This might be a little cleaner with strings.Cut. Give it a try and see if you like it better.

Reply:
`string.Cut` cuts after first separator, we have to cut after last